src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -62,41 +93,192 @@ private string GetWorkersRootDirPath(IConfiguration configuration)
+        /// Feature flag takes precedence over hosting config options.
+        /// </summary>
+        internal bool IsDynamicWorkerResolutionEnabled(string workerRuntime, HashSet<string> workersAvailableForResolution, bool isPlaceholderModeEnabled, bool isMultiLanguageEnv)
+        {
+            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDynamicWorkerResolution, _environment) || workersAvailableForResolution.Count == 0)
             {
-                return _configuration;
+                return false;
             }
 
-            return null;
+            if (!isPlaceholderModeEnabled && !isMultiLanguageEnv && !string.IsNullOrWhiteSpace(workerRuntime))
+            {
+                return workersAvailableForResolution.Contains(workerRuntime);
+            }
+
+            return true;
         }
 
-        private string GetConfigurationSectionValue(IConfiguration configuration, string configurationSource, string requiredSection)
+        /// <summary>
+        /// Returns a dictionary of worker names to sets of ignored versions, parsed from hosting config options.
+        /// Output format: { worker: { hashset of versions to be ignored }}.
+        /// Sample output: {"java": {"2.19.0", "2.18.0"}, "dotnet-isolated": {"1.0.0"}}.
+        /// </summary>
+        internal Dictionary<string, HashSet<Version>> GetIgnoredWorkerVersions()
         {
-            var section = configuration?.GetSection(requiredSection);
+            // Example value of ignoredWorkerVersions: "Worker1Name:Version1|Worker1Name:Version2|Worker2Name:Version1|Worker3Name:Version1".
+            string ignoredWorkerVersions = _functionsHostingConfigOptions.Value.IgnoredWorkerVersions;
+
+            if (string.IsNullOrWhiteSpace(ignoredWorkerVersions))
+            {
+                return new(StringComparer.OrdinalIgnoreCase);
+            }
 
-            if (!string.IsNullOrEmpty(section?.Value))
+            var ignoredVersions = ignoredWorkerVersions.Split('|', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            var ignoredVersionsOut = new Dictionary<string, HashSet<Version>>(capacity: ignoredVersions.Length, comparer: StringComparer.OrdinalIgnoreCase);
+
+            foreach (string ignoredVersion in ignoredVersions)
             {
-                _logger.LogTrace("Found configuration section '{requiredSection}' in '{configurationSource}'.", requiredSection, configurationSource);
-                return section.Value;
+                string[] workerVersionParts = ignoredVersion.Split(':', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+                if (workerVersionParts.Length != 2)
+                {
+                    _logger.LogTrace("Skipping '{ignoredVersion}' due to invalid format for ignored worker version. Expected format is 'WorkerName:Version'.", ignoredVersion);
```

Comment:
fyi -- same thing as elsewhere (and check everywhere else) -- but LogTrace won't show in kusto